reactivesocket-core/src/main/java/io/reactivesocket/util/PayloadImpl.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ * <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ *  the License. You may obtain a copy of the License at
+ *  <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  <p>
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivesocket.util;
+
+import io.reactivesocket.Frame;
+import io.reactivesocket.Payload;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+
+/**
+ * An implementation of {@link Payload}
+ */
+public class PayloadImpl implements Payload {
+
+    private final ByteBuffer data;
+    private final ByteBuffer metadata;
+
+    public PayloadImpl(String data) {
```

Comment:
I'd suggest dropping the String constructors as you already have the static factory methods.  These 4 constructors make it seem like Payload carries Strings or byte[]+Charset